addons/test_mail/tests/test_mail_alias.py

```diff
@@ -280,9 +333,83 @@ def test_res_company_creation_alias_domain(self):
         })
         company.flush_recordset()
         self.assertEqual(
-            company.catchall_formatted,
-            formataddr((company.name, f'{self.alias_catchall}@{self.alias_domain}'))
-        )
+            company.alias_domain_id, self.mail_alias_domain,
+            'Default alias domain: sequence based')
+
+        # respect forced value
+        company = self.env['res.company'].create({
+            'alias_domain_id': self.mail_alias_domain_c2.id,
+            'email': '"Yet Another Company" <yet.another.company@test.embed.mycompany.com>',
+            'name': 'Yet Another Company',
+        })
+        self.assertEqual(company.alias_domain_id, self.mail_alias_domain_c2)
+
+
+@tagged('mail_gateway', 'mail_alias', 'multi_company')
+class TestMailAliasDomain(TestMailAliasCommon):
+
+    @users('admin')
+    def test_alias_domain_name_unique(self):
+        """ Test uniqueness of alias domain names, as we should not have email
+        clashes. """
+        alias_domain = self.mail_alias_domain.with_env(self.env)
+        new_alias_domain = alias_domain.copy({'name': 'Copy'})
+
+        with mute_logger('odoo.sql_db'), self.assertRaises(psycopg2.errors.UniqueViolation), self.cr.savepoint():
+            self.env['mail.alias.domain'].create({'name': alias_domain.name})
+        with mute_logger('odoo.sql_db'), self.assertRaises(psycopg2.errors.UniqueViolation), self.cr.savepoint():
+            new_alias_domain.write({'name': alias_domain.name})
+        with mute_logger('odoo.sql_db'), self.assertRaises(psycopg2.errors.UniqueViolation), self.cr.savepoint():
+            alias_domain.copy()
```

Comment:
Shouldn't we override the copy then, so that users can click on duplicate without having the interface crash? (by maybe just adding _copy in the name or something).